### PR TITLE
fix: ensure correct ordering of sendSync w.r.t. send

### PR DESCRIPTION
**Pull request by nornagon**
#### Description of Change
#18449 introduced a potential race between `send` and `sendSync` by pushing `sendSync` calls to a separate mojo pipe, which is the cause of the recent test flake in `remote module` tests. This change rearranges the calls so `send` and family all happen on the same pipe, and only `invoke` happens on a different pipe, which it must do because it has a callback with a result (which can't be pending when `sendSync` is called, or else DCHECK). There are no ordering concerns currently w/ `invoke` because we don't really use it, but in future we probably want to guarantee that `send` and `invoke` share an ordering, which will require some further trickery.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [ ] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: no-notes

### CODE REVIEW

File: atom/renderer/api/atom_api_renderer_ipc.cc [hunk 1]
```diff
@@ -55,49 +61,47 @@ class IPCRenderer : public mate::Wrappable<IPCRenderer> {
...
     return mate::CreateHandle(isolate, new IPCRenderer(isolate));
   }
 
-  void Send(mate::Arguments* args,
-            bool internal,
+  void Send(bool internal,
             const std::string& channel,
             const base::ListValue& arguments) {
-    electron_browser_ptr_->Message(internal, channel, arguments.Clone());
+    electron_browser_ptr_->get()->Message(internal, channel, arguments.Clone());
```
Comment (deepak1556) [inline atom/renderer/api/atom_api_renderer_ipc.cc]: Now that the interface ptr is created on the background runner, which is a different sequence to that of the main runner. Is it okay to call methods on this ptr in the main sequence without calling `PassInterface` ? well that would technically create a new pipe which would defeat the purpose. Just to make sure is this a safe operation ?

Comment (nornagon) [reply atom/renderer/api/atom_api_renderer_ipc.cc]: The type of this pointer changed to `ThreadSafeElectronBrowserPtr`, whose _raison d'etre_ is to allow calling methods from multiple threads. This doesn't create any new pipes.

**Comment by nornagon** [comment]
I'm merging this ahead of review because it fixes a very annoying flake on master, but I'm happy to incorporate feedback from @zcbenz or anything further from @deepak1556 :)

File: atom/renderer/api/atom_api_renderer_ipc.cc (hunk 1 above)
Comment (deepak1556) [reply atom/renderer/api/atom_api_renderer_ipc.cc]: Ah okay it uses `mojo/public/cpp/bindings/thread_safe_interface_ptr.h` , wasn't aware of this helper.

Comment (nornagon) [reply atom/renderer/api/atom_api_renderer_ipc.cc]: I wasn't either until I asked in #mojo on the Chromium Slack :)

**Comment by zcbenz** [comment]
Not sure if it is related, but after rebasing to this change, I'm constantly seeing "Object has been destroyed" error.

Steps:
1. Checkout https://github.com/electron/electron/pull/18483
2. Run `test -g crashReporter`
3. Switch to another window and then switch back

Screen Shot 2019-06-05 at 17 15 28

I have not tested the master branch yet.

**Comment by miniak** [comment]
@nornagon should it be backported to 6-0-x?

**Pull request merged**